src/index.js

```diff
@@ -21,15 +23,18 @@ export default class HtmlWebpackInjectPlugin {
 
   apply = compiler => {
     compiler.hooks.compilation.tap('HtmlWebpackInjectPlugin', compilation => {
-      compilation.hooks.htmlWebpackPluginAlterAssetTags.tapAsync(
-        'HtmlWebpackInjectPlugin',
-        (htmlPluginData, cb) => {
-          htmlPluginData[this.parent] = this.prepend
-            ? this.assets.concat(htmlPluginData[this.parent])
-            : htmlPluginData[this.parent].concat(this.assets)
-          return cb(null, htmlPluginData)
-        }
-      )
+      const hooks =
+        HtmlWebpackPlugin.version === 4
```

Comment:
how about major upgrade?

Reply:
Many plugins still not compatible with v4, people may want to use them at the same time. But I'm ok with that, it is your choice.